From ddbd9ee970752d8440010d5e2edd59bd7bd0c91d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?D=2E=20Andrei=20M=C4=83ce=C8=99?= <dmaces@nd.edu>
Date: Wed, 24 Aug 2016 12:58:43 -0400
Subject: [PATCH 07/18] board: Localize COMMON_GLOBAL_CFLAGS to respective
 modules

* -DENHANCED_DOMX already localized, so remove it here
* -DUSE_TI_LIBION was propagated to modules/pvr-source
* -DOMAP_ENHANCEMENT moved to camera & test/CameraHal
* -DOMAP_ENHANCEMENT_BURST_CAPTURE, same as above
* -DOMAP_ENHANCEMENT_S3D moved to camera & hwc
* -DOMAP_ENHANCEMENT_CPCAM to TI_CAMERAHAL_COMMON_CFLAGS
* -DOMAP_ENHANCEMENT_VTC to camera & domx/omx_proxy_component
* -DUSE_ITTIAM_AAC moved to domx/omx_core
* -DOMAP_ENHANCEMENT_MULTIGPU is NOOP

Change-Id: Ib780ede012bcb5943643fca8864aba8415dc956e
---
 BoardConfigCommon.mk                               | 34 ----------------------
 camera/Android.mk                                  | 19 ++++++++++++
 domx/domx/Android.mk                               |  1 +
 domx/omx_core/Android.mk                           |  4 +++
 domx/omx_proxy_component/Android.mk                |  5 ++++
 hwc/Android.mk                                     |  7 ++++-
 libion_ti/Android.mk                               |  4 ++-
 .../build/linux2/common/android/arch_common.mk     |  4 +++
 test/CameraHal/Android.mk                          |  9 ++++++
 9 files changed, 51 insertions(+), 36 deletions(-)

diff --git a/BoardConfigCommon.mk b/BoardConfigCommon.mk
index fdcadad..784b8c1 100644
--- a/BoardConfigCommon.mk
+++ b/BoardConfigCommon.mk
@@ -38,45 +38,11 @@ BOARD_USE_TI_ENHANCED_DOMX := true
 BOARD_USE_TI_DUCATI_H264_PROFILE := true
 BOARD_USE_TI_DOMX_LOW_SECURE_HEAP := true
 BOARD_USE_TI_LIBION ?= true
-COMMON_GLOBAL_CFLAGS += -DENHANCED_DOMX
 ENHANCED_DOMX := true
 TARGET_SPECIFIC_HEADER_PATH += $(OMAP4_NEXT_FOLDER)/domx/omx_core/inc
 BOARD_USE_TI_CUSTOM_DOMX := true
 DOMX_PATH := $(OMAP4_NEXT_FOLDER)/domx
 
-ifeq ($(BOARD_USE_TI_LIBION),true)
-    COMMON_GLOBAL_CFLAGS += -DUSE_TI_LIBION
-endif
-
-ifdef OMAP_ENHANCEMENT
-    COMMON_GLOBAL_CFLAGS += -DOMAP_ENHANCEMENT -DTARGET_OMAP4 -DFORCE_SCREENSHOT_CPU_PATH
-endif
-
-ifdef OMAP_ENHANCEMENT_BURST_CAPTURE
-    COMMON_GLOBAL_CFLAGS += -DOMAP_ENHANCEMENT_BURST_CAPTURE
-endif
-
-ifdef OMAP_ENHANCEMENT_S3D
-    COMMON_GLOBAL_CFLAGS += -DOMAP_ENHANCEMENT_S3D
-endif
-
-ifdef OMAP_ENHANCEMENT_CPCAM
-    COMMON_GLOBAL_CFLAGS += -DOMAP_ENHANCEMENT_CPCAM
-    PRODUCT_MAKEFILES += $(LOCAL_DIR)/sdk_addon/ti_omap_addon.mk
-endif
-
-ifdef OMAP_ENHANCEMENT_VTC
-    COMMON_GLOBAL_CFLAGS += -DOMAP_ENHANCEMENT_VTC
-endif
-
-ifdef USE_ITTIAM_AAC
-    COMMON_GLOBAL_CFLAGS += -DUSE_ITTIAM_AAC
-endif
-
-ifdef OMAP_ENHANCEMENT_MULTIGPU
-    COMMON_GLOBAL_CFLAGS += -DOMAP_ENHANCEMENT_MULTIGPU
-endif
-
 # Bootanimation
 TARGET_BOOTANIMATION_PRELOAD := false
 TARGET_BOOTANIMATION_TEXTURE_CACHE := false
diff --git a/camera/Android.mk b/camera/Android.mk
index 0346ceb..ba0fe62 100755
--- a/camera/Android.mk
+++ b/camera/Android.mk
@@ -54,6 +54,22 @@ ifeq ($(ENHANCED_DOMX),true)
     TI_CAMERAHAL_COMMON_CFLAGS += -DENHANCED_DOMX
 endif
 
+ifdef OMAP_ENHANCEMENT
+    TI_CAMERAHAL_COMMON_CFLAGS += -DOMAP_ENHANCEMENT
+endif
+
+ifdef OMAP_ENHANCEMENT_BURST_CAPTURE
+    TI_CAMERAHAL_COMMON_CFLAGS += -DOMAP_ENHANCEMENT_BURST_CAPTURE
+endif
+
+ifdef OMAP_ENHANCEMENT_S3D
+    TI_CAMERAHAL_COMMON_CFLAGS += -DOMAP_ENHANCEMENT_S3D
+endif
+
+ifdef OMAP_ENHANCEMENT_VTC
+    TI_CAMERAHAL_COMMON_CFLAGS += -DOMAP_ENHANCEMENT_VTC
+endif
+
 ifdef ARCH_ARM_HAVE_NEON
     TI_CAMERAHAL_COMMON_CFLAGS += -DARCH_ARM_HAVE_NEON
 endif
@@ -171,6 +187,7 @@ TI_CAMERAHAL_COMMON_SHARED_LIBRARIES := \
 
 ifeq ($(BOARD_USE_TI_LIBION),true)
 TI_CAMERAHAL_COMMON_SHARED_LIBRARIES += libion_ti
+TI_CAMERAHAL_COMMON_CFLAGS += -DUSE_TI_LIBION
 else
 TI_CAMERAHAL_COMMON_SHARED_LIBRARIES += libion
 TI_CAMERAHAL_COMMON_SRC +=  ../libion/ion_ti_custom.c
@@ -186,6 +203,8 @@ TI_CAMERAHAL_OMX_SHARED_LIBRARIES := \
 ifdef OMAP_ENHANCEMENT_CPCAM
 TI_CAMERAHAL_COMMON_STATIC_LIBRARIES += \
     libcpcamcamera_client
+TI_CAMERAHAL_COMMON_CFLAGS += -DOMAP_ENHANCEMENT_CPCAM
+PRODUCT_MAKEFILES += $(LOCAL_DIR)/sdk_addon/ti_omap_addon.mk
 endif
 
 
diff --git a/domx/domx/Android.mk b/domx/domx/Android.mk
index e8b039b..dbd939b 100644
--- a/domx/domx/Android.mk
+++ b/domx/domx/Android.mk
@@ -40,6 +40,7 @@ LOCAL_SHARED_LIBRARIES := \
 
 ifeq ($(BOARD_USE_TI_LIBION),true)
 LOCAL_SHARED_LIBRARIES += libion_ti
+LOCAL_CFLAGS += -DUSE_TI_LIBION
 else
 LOCAL_SHARED_LIBRARIES += libion
 LOCAL_SRC_FILES += ../../libion/ion_ti_custom.c
diff --git a/domx/omx_core/Android.mk b/domx/omx_core/Android.mk
index f35eb2b..3308e79 100644
--- a/domx/omx_core/Android.mk
+++ b/domx/omx_core/Android.mk
@@ -16,6 +16,10 @@ LOCAL_SHARED_LIBRARIES := \
     liblog \
     libmm_osal
 
+ifdef USE_ITTIAM_AAC
+LOCAL_CFLAGS += -DUSE_ITTIAM_AAC
+endif
+
 LOCAL_CFLAGS += -DSTATIC_TABLE -D_Android -DCHECK_SECURE_STATE
 LOCAL_MODULE:= libOMX_Core
 LOCAL_MODULE_TAGS:= optional
diff --git a/domx/omx_proxy_component/Android.mk b/domx/omx_proxy_component/Android.mk
index 2dc71ac..00bac1a 100644
--- a/domx/omx_proxy_component/Android.mk
+++ b/domx/omx_proxy_component/Android.mk
@@ -105,12 +105,17 @@ LOCAL_SHARED_LIBRARIES := \
 
 ifeq ($(BOARD_USE_TI_LIBION),true)
 LOCAL_SHARED_LIBRARIES += libion_ti
+LOCAL_CFLAGS += -DUSE_TI_LIBION
 else
 LOCAL_SHARED_LIBRARIES += libion
 LOCAL_SRC_FILES += ../../libion/ion_ti_custom.c
 LOCAL_C_INCLUDES += $(HARDWARE_TI_OMAP4_BASE)/libion
 endif
 
+ifdef OMAP_ENHANCEMENT_VTC
+LOCAL_CFLAGS += -DOMAP_ENHANCEMENT_VTC
+endif
+
 LOCAL_CFLAGS += -DTMS32060 -D_DB_TIOMAP -DSYSLINK_USE_SYSMGR -DSYSLINK_USE_LOADER
 LOCAL_CFLAGS += -D_Android -DSET_STRIDE_PADDING_FROM_PROXY -DANDROID_QUIRK_CHANGE_PORT_VALUES -DUSE_ENHANCED_PORTRECONFIG
 LOCAL_CFLAGS += -DANDROID_QUIRK_LOCK_BUFFER -DUSE_ION
diff --git a/hwc/Android.mk b/hwc/Android.mk
index d2d6b0b..af7e683 100644
--- a/hwc/Android.mk
+++ b/hwc/Android.mk
@@ -8,21 +8,26 @@ LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/../vendor/lib/hw
 LOCAL_SHARED_LIBRARIES := liblog libEGL libcutils libutils libhardware libhardware_legacy libz
 
 LOCAL_SRC_FILES := hwc.c rgz_2d.c dock_image.c sw_vsync.c display.c
+LOCAL_CFLAGS := -DLOG_TAG=\"ti_hwc\" -Wall -Werror
 
 ifeq ($(BOARD_USE_TI_LIBION),true)
 LOCAL_SHARED_LIBRARIES += libion_ti
+LOCAL_CFLAGS += -DUSE_TI_LIBION
 else
 LOCAL_SHARED_LIBRARIES += libion
 LOCAL_SRC_FILES += ../libion/ion_ti_custom.c
 LOCAL_C_INCLUDES += $(HARDWARE_TI_OMAP4_BASE)/libion
 endif
 
+ifdef OMAP_ENHANCEMENT_S3D
+LOCAL_CFLAGS += -DOMAP_ENHANCEMENT_S3D
+endif
+
 LOCAL_STATIC_LIBRARIES := libpng
 
 LOCAL_MODULE_TAGS := optional
 
 LOCAL_MODULE := hwcomposer.omap4
-LOCAL_CFLAGS := -DLOG_TAG=\"ti_hwc\" -Wall -Werror
 LOCAL_C_INCLUDES += external/libpng external/zlib
 
 LOCAL_C_INCLUDES += \
diff --git a/libion_ti/Android.mk b/libion_ti/Android.mk
index b87f8ee..5516566 100644
--- a/libion_ti/Android.mk
+++ b/libion_ti/Android.mk
@@ -5,7 +5,7 @@ LOCAL_PATH:= $(call my-dir)
 
 include $(CLEAR_VARS)
 LOCAL_SRC_FILES := ion.c
-LOCAL_CFLAGS := -Wall -Werror
+LOCAL_CFLAGS := -Wall -Werror -DUSE_TI_LIBION
 LOCAL_MODULE := libion_ti
 LOCAL_MODULE_TAGS := optional
 LOCAL_SHARED_LIBRARIES := liblog
@@ -13,6 +13,7 @@ include $(BUILD_SHARED_LIBRARY)
 
 include $(CLEAR_VARS)
 LOCAL_SRC_FILES := ion.c ion_test.c
+LOCAL_CFLAGS := -DUSE_TI_LIBION
 LOCAL_MODULE := ion_ti_test
 LOCAL_MODULE_TAGS := optional tests
 LOCAL_SHARED_LIBRARIES := liblog
@@ -20,6 +21,7 @@ include $(BUILD_EXECUTABLE)
 
 include $(CLEAR_VARS)
 LOCAL_SRC_FILES := ion.c ion_test_2.c
+LOCAL_CFLAGS := -DUSE_TI_LIBION
 LOCAL_MODULE := ion_ti_test2
 LOCAL_MODULE_TAGS := optional tests
 LOCAL_SHARED_LIBRARIES := liblog
diff --git a/pvr-source/eurasiacon/build/linux2/common/android/arch_common.mk b/pvr-source/eurasiacon/build/linux2/common/android/arch_common.mk
index 23f9175..147508f 100644
--- a/pvr-source/eurasiacon/build/linux2/common/android/arch_common.mk
+++ b/pvr-source/eurasiacon/build/linux2/common/android/arch_common.mk
@@ -54,6 +54,10 @@ SYS_CFLAGS := \
  -I$(ANDROID_ROOT)/hardware/libhardware/include \
  -I$(ANDROID_ROOT)/external/openssl/include
 
+ifeq ($(USE_TI_LIBION),1)
+SYS_CFLAGS += -DUSE_TI_LIBION
+endif
+
 SYS_EXE_LDFLAGS := \
  -Bdynamic -nostdlib -Wl,-dynamic-linker,/system/bin/linker \
  -lc -ldl -lcutils
diff --git a/test/CameraHal/Android.mk b/test/CameraHal/Android.mk
index 60a6b92..aa37971 100644
--- a/test/CameraHal/Android.mk
+++ b/test/CameraHal/Android.mk
@@ -49,6 +49,14 @@ ifeq ($(TARGET_BOARD_PLATFORM),omap4)
     LOCAL_CFLAGS += -DTARGET_OMAP4
 endif
 
+ifdef OMAP_ENHANCEMENT
+    LOCAL_CFLAGS += -DOMAP_ENHANCEMENT
+endif
+
+ifdef OMAP_ENHANCEMENT_BURST_CAPTURE
+    LOCAL_CFLAGS += -DOMAP_ENHANCEMENT_BURST_CAPTURE
+endif
+
 include $(BUILD_HEAPTRACKED_EXECUTABLE)
 
 
@@ -73,6 +81,7 @@ LOCAL_SHARED_LIBRARIES:= \
 
 ifeq ($(BOARD_USE_TI_LIBION),true)
 LOCAL_SHARED_LIBRARIES += libion_ti
+LOCAL_CFLAGS += -DUSE_TI_LIBION
 else
 LOCAL_SHARED_LIBRARIES += libion
 LOCAL_SRC_FILES += ../../libion/ion_ti_custom.c
-- 
2.7.4

